### PR TITLE
Create Item class 

### LINKED ISSUE #28: Create Item Class
## The Item instance offers the following methods:

- id - returns the integer id of the item
- name - returns the name of the item
- description - returns the description of the item
- unit_price - returns the price of the item formatted as a BigDecimal
- created_at - returns a Time instance for the date the item was first created
- updated_at - returns a Time instance for the date the item was last modified
- merchant_id - returns the integer merchant id of the item
- It also offers the following method:

- unit_price_to_dollars - returns the price of the item in dollars formatted as a Float
We create an instance like this:
```rb
i = Item.new({
  :id          => 1,
  :name        => "Pencil",
  :description => "You can use it to write things",
  :unit_price  => BigDecimal.new(10.99,4),
  :created_at  => Time.now,
  :updated_at  => Time.now,
  :merchant_id => 2
})
```

Effort Points 2

### PR DESCRIPTION
**Pull request by NoahZinter**
### Changes

- Description of changes
Creates item class, item_class_spec. 

### Requested review on...
- Specifics on, logic, naming schemes, etc.
Lines 85--113, We tested whether the return was a Time class, but would need a Mock to test that it's the same time. Should we Mock it? 

### Checklist
- [x ] Tests pass

closes #28

### CODE REVIEW

File: lib/item.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def initialize(details)
+    @id = details[:id]
+    @name = details[:name]
+    @description = details[:description]
+    @unit_price = details[:unit_price]
+    @created_at = details[:created_at]
+    @updated_at = details[:updated_at]
+    @merchant_id = details[:merchant_id]
+  end
+
```
Comment (sami-p) [inline lib/item.rb:19]: Thanks for matching indentation for consistency.

File: spec/item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        :name => "Pencil",
+        :description => "You can use it to write things",
+        :unit_price => BigDecimal.new(10.99,4),
+        :created_at => time,
+        :updated_at => time,
+        :merchant_id => 2
+      }
+      item = Item.new(details)
+
+      expect(item.created_at).to eq(time)
```
Comment (sami-p) [inline spec/item_spec.rb:99]: Will want to remove the parenthesis around 'time'

File: spec/item_spec.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        :name => "Pencil",
+        :description => "You can use it to write things",
+        :unit_price => BigDecimal.new(10.99,4),
+        :created_at => time,
+        :updated_at => time,
+        :merchant_id => 2
+      }
+      item = Item.new(details)
+
+      expect(item.updated_at).to eq(time)
```
Comment (sami-p) [inline spec/item_spec.rb:116]: Remove parenthesis around 'time'

File: spec/item_spec.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        :name => "Pencil",
+        :description => "You can use it to write things",
+        :unit_price => BigDecimal.new(10.99,4),
+        :created_at => Time.now,
+        :updated_at => Time.now,
+        :merchant_id => 2
+      }
+      item = Item.new(details)
+
+      expect(item.unit_price_to_dollars).to eq(10.99)
```
Comment (sami-p) [inline spec/item_spec.rb:150]: I think we'll want to remove the parenthesis around '10.99' too, correct?

**Review by sami-p — APPROVED** [review]
Super clear and thorough work! I marked the lines where we may need to remove parenthesis from the end of the expect statements in the spec file.

File: spec/item_spec.rb [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      item = Item.new(details)
+
+      expect(item.unit_price).to eq BigDecimal.new(10.99,4)
+      expect(item.unit_price).is_a? BigDecimal
+    end
+
+    it 'has a created_at Time' do
+      time = Time.now
+      details = {
+        :id => 1,
```
Comment (redferret) [inline spec/item_spec.rb]: I think I know how to fix this. 
The rockets need to be changed to `:` and the keys without the `:`
```rb
details = {
  id: 1,
  name: 'Pencil'
}
```
That is the newest syntax for Ruby hashes

**Review by redferret — CHANGES_REQUESTED** [review]
We can clear a lot of Hound out by updating the syntax for the hashes to the following:
Instead of this
```rb
hash = {
  :id => 10
}
```
Make it to this
```rb
hash = {
  id: 10
}
```
and change all strings to single quotes not double quotes.
That will clear a lot of Hound out

**Review by redferret — APPROVED** [review]
Looks great!

**Pull request merged**